cli/src/cmd/cast/wallet/mod.rs

```diff
@@ -55,15 +56,11 @@ pub enum WalletSubcommands {
         wallet: Wallet,
     },
 
-    /// Sign a message.
+    /// Sign a message or typed data.
     #[clap(visible_alias = "s")]
     Sign {
-        /// The message to sign.
-        ///
-        /// Messages starting with 0x are expected to be hex encoded,
-        /// which get decoded before being signed.
-        message: String,
-
+        #[clap(subcommand)]
+        command: Option<SignType>,
```

Comment:
oh now that I had a closer look, this enum is a breaking change because signing text now requires `message` command?

This is a bit suboptimal, 
I think we can add this feature in a nonbreaking way by adding a `--data` bool flag which interprets the message as typed data?

wdyt?

Reply:
Yeah, that seems reasonable. 

Wdyt about the from_file param? Should I get rid of that and users and load their JSON into the shell however they see fit. E.g.
`cast wallet sign --data "$(cat ./data.json)"`

Reply:
yeah that should work!
but for typedata, I think we can keep the `param is file` check, doesn't hurt